legion_core/src/world.rs

```diff
@@ -712,6 +717,56 @@ impl World {
         unsafe { self.get_component_mut_unchecked(entity) }
     }
 
+    pub fn get_all_components<'a, T: Component>(&'a self) -> RefMapSet<'a, Vec<&'a T>> {
```

Comment:
doc string

Reply:
This PR includes some changes that are redundant with #105, I'm hoping we can get that one in first and rebase this on top of it. #105 has doc strings for these functions.